wire-ios/Wire-iOS/Sources/Analytics/TrackingManager/TrackingManager.swift

```diff
@@ -48,21 +48,36 @@ final class TrackingManager: NSObject, TrackingInterface {
             ExtensionSettings.shared.disableAnalyticsSharing
         }
         set {
-            ExtensionSettings.shared.disableAnalyticsSharing = newValue
-            AVSFlowManager.getInstance()?.setEnableMetrics(!newValue)
-
-            do {
-                if newValue {
-                    let disableUseCase = try sessionManager.makeDisableAnalyticsSharingUseCase()
-                    disableUseCase.invoke()
+            self.showAnalyticsConsentAlert { userConsented in
+                if userConsented {
+                    self.updateAnalyticsSharing(!newValue)
                 } else {
-                    let enableUseCase = try sessionManager.makeEnableAnalyticsSharingUseCase()
-                    enableUseCase.invoke()
+                    // User rejected, so we keep analytics disabled
+                    self.updateAnalyticsSharing(true)
                 }
-            } catch {
-                print("Failed to toggle analytics sharing: \(error)")
             }
         }
     }
 
+    private func updateAnalyticsSharing(_ disable: Bool) {
```

Comment:
I suggest to give the argument a name because it's hard to know what the bool means from the call site. E.g `updateAnalyticsSharing(true)`, what does true mean without looking at the implementation?

Reply:
Done: https://github.com/wireapp/wire-ios/pull/1767/commits/4d057e135cfeef971c25594ac2e148f186b7e24f